### PR TITLE
Operator is aware about changes on underlying ScaledObject

### LINKED ISSUE #900: Operator doesn't watch changes on generated sub-resources
### Report

Currently, once the operator has parsed the HTTPScaledObject and generated the ScaledObject, it doesn't watch the ScaledObject until the operator is restarted. This isn't worth because the (add-on) operator is the owner of the ScaledObject and it should be always aware about any relevant change on it (even about deletions to recreate it)

### Expected Behavior

The operator is the only owner of the generated ScaledObject and replaces any manual done change

### Actual Behavior

The ScaledObject is created and ignored until a restart

### PR DESCRIPTION
**Pull request by JorTurFer**
### Checklist

- [x] Commits are signed with Developer Certificate of Origin (DCO)
- [x] Changelog has been updated and is aligned with our [changelog requirements](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#Changelog)

Fixes #900

### CODE REVIEW

File: config/crd/bases/http.keda.sh_httpscaledobjects.yaml [hunk 1]
```diff
@@ -155,12 +155,6 @@ spec:
                     type:
                       description: Type of condition
                       enum:
-                      - Created
-                      - Terminated
-                      - Error
-                      - Pending
-                      - Terminating
-                      - Unknown
```
Comment (JorTurFer) [inline config/crd/bases/http.keda.sh_httpscaledobjects.yaml:163]: Currently, we have too many conditions without a huge sense, I've just removed all except `Ready` and I plan to give some love in the future to rethink this to make it useful for users

File: operator/apis/http/v1alpha1/condition_types.go [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+const (
+	ErrorCreatingAppScaledObject    HTTPScaledObjectConditionReason = "ErrorCreatingAppScaledObject"
+	AppScaledObjectCreated          HTTPScaledObjectConditionReason = "AppScaledObjectCreated"
+	TerminatingResources            HTTPScaledObjectConditionReason = "TerminatingResources"
+	AppScaledObjectTerminated       HTTPScaledObjectConditionReason = "AppScaledObjectTerminated"
+	AppScaledObjectTerminationError HTTPScaledObjectConditionReason = "AppScaledObjectTerminationError"
+	PendingCreation                 HTTPScaledObjectConditionReason = "PendingCreation"
+	HTTPScaledObjectIsReady         HTTPScaledObjectConditionReason = "HTTPScaledObjectIsReady"
+)
```
Comment (JorTurFer) [inline operator/apis/http/v1alpha1/condition_types.go:29]: Same as above, I plan to revisit this in the future to clean these conditions to just the required

File: operator/apis/http/v1alpha1/condition_types.go [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+}
+
+func (c Conditions) getCondition(conditionType HTTPScaledObjectCreationStatus) HTTPScaledObjectCondition {
+	for i := range c {
+		if c[i].Type == conditionType {
+			return c[i]
+		}
+	}
+	return HTTPScaledObjectCondition{}
+}
```
Comment (JorTurFer) [inline operator/apis/http/v1alpha1/condition_types.go:88]: I've ported these functions from KEDA to standardize a bit the code. I plan to open another PR unifying the code that updates the status with KEDA

File: operator/controllers/http/httpscaledobject_controller.go [hunk 4]
```diff
@@ -146,7 +148,20 @@ func (r *HTTPScaledObjectReconciler) Reconcile(ctx context.Context, req ctrl.Req
...
+				predicate.GenerationChangedPredicate{},
+				util.HTTPScaledObjectReadyConditionPredicate{},
+			),
+		)).
+		// Trigger a reconcile only when the ScaledObject spec,label or annotation changes.
+		// Ignore updates to ScaledObject status
+		Owns(&kedav1alpha1.ScaledObject{}, builder.WithPredicates(
+			predicate.Or(
+				predicate.LabelChangedPredicate{},
+				predicate.AnnotationChangedPredicate{},
```
Comment (JorTurFer) [inline operator/controllers/http/httpscaledobject_controller.go:162]: I'm not 100% sure if we should be aware of changes on labels, but I think that we have to be for annotations as we propagate them to the ScaledObject for pausing actions

File: config/crd/bases/http.keda.sh_httpscaledobjects.yaml (hunk 1 above)
Comment (JorTurFer) [reply config/crd/bases/http.keda.sh_httpscaledobjects.yaml:163]: https://github.com/kedacore/http-add-on/issues/902

File: operator/apis/http/v1alpha1/condition_types.go (hunk 3 above)
Comment (JorTurFer) [reply operator/apis/http/v1alpha1/condition_types.go:88]: https://github.com/kedacore/http-add-on/issues/902

File: CHANGELOG.md [hunk 5]
```diff
@@ -29,7 +29,7 @@ This changelog keeps track of work items that have been completed and are ready
 
 ### Fixes
 
-- **General**: TODO ([#TODO](https://github.com/kedacore/http-add-on/issues/TODO))
+- **General**: Operator is aware about changes on underlying ScaledObject ([#900](https://github.com/kedacore/http-add-on/issues/900))
```
Comment (tomkerkhove) [inline CHANGELOG.md:32]:
```suggestion
- **General**: Ensure operator is aware about changes on underlying ScaledObject ([#900](https://github.com/kedacore/http-add-on/issues/900))
```

**Review by tomkerkhove — APPROVED** [review]

File: operator/controllers/http/httpscaledobject_controller.go (hunk 4 above)
Comment (zroubalik) [reply operator/controllers/http/httpscaledobject_controller.go:162]: Agree

**Pull request merged**